skaffold/src/main/kotlin/com/google/container/tools/skaffold/SkaffoldConfigurationDetector.kt

```diff
@@ -95,42 +94,60 @@ class SkaffoldConfigurationDetector(val project: Project) : ProjectComponent {
 
     @VisibleForTesting
     fun addSkaffoldDevConfiguration(skaffoldFilePath: String) {
-        val skaffoldDevSettings = SkaffoldDevConfiguration(
-            project,
-            SkaffoldDevConfigurationFactory(SkaffoldRunConfigurationType()),
-            message("skaffold.run.config.dev.default.name")
+        createRunConfigurationFromSettings(
+            SkaffoldDevConfigurationFactory.DEV_ID,
+            message("skaffold.run.config.dev.default.name"),
+            skaffoldFilePath
         )
-        skaffoldDevSettings.skaffoldConfigurationFilePath = skaffoldFilePath
-
-        createRunConfigurationFromSettings(skaffoldDevSettings)
     }
 
     @VisibleForTesting
     fun addSkaffoldRunConfiguration(skaffoldFilePath: String) {
-        val skaffoldRunSettings = SkaffoldSingleRunConfiguration(
-            project,
-            SkaffoldSingleRunConfigurationFactory(SkaffoldRunConfigurationType()),
-            message("skaffold.run.config.run.default.name")
+        createRunConfigurationFromSettings(
+            SkaffoldSingleRunConfigurationFactory.RUN_ID,
+            message("skaffold.run.config.run.default.name"),
+            skaffoldFilePath
         )
-        skaffoldRunSettings.skaffoldConfigurationFilePath = skaffoldFilePath
-
-        createRunConfigurationFromSettings(skaffoldRunSettings)
     }
 
     /** Creates a run configuration from the given settings and selects it in run combobox */
-    private fun createRunConfigurationFromSettings(runConfiguration: RunConfiguration) {
-        val runnerAndConfigSettings = RunnerAndConfigurationSettingsImpl(
-            RunManagerImpl.getInstanceImpl(project),
-            runConfiguration
-        )
+    private fun createRunConfigurationFromSettings(
+        factoryId: String,
+        name: String,
+        skaffoldFilePath: String
+    ) {
+
+        val factory = findConfigurationFactoryById(factoryId)
+        if (factory == null) {
+            Logger.getInstance(SkaffoldConfigurationDetector::class.java)
+                .warn("Skaffold configuration factory not found, plugin install is corrupted.")
```

Comment:
should we make this an error instead of a warn since its truly something we don't expect to ever happen?

Reply:
I think so, yes, that would indicate some completely unexpected corruption / misconfg.